src/components/FlashCardApp.tsx

```diff
@@ -0,0 +1,156 @@
+import {
+    Alert,
+    AlertDescription,
+    AlertIcon,
+    AlertTitle,
+    Select,
+    Stack,
+    VStack,
+    Text,
+} from "@chakra-ui/react";
+import { Card, Deck, User } from "../utils/types";
+import { Flashcard } from "./Flashcard";
+import { useEffect, useState } from "react";
+import axios from "axios";
+import AppOptions from "./AppOptions";
+import { useImmer } from "use-immer";
+import { baseUrl } from "../utils/baseUrl";
+
+interface FlashCardAppProps {
+    user: User;
+}
+
+export function FlashCardApp({ user }: FlashCardAppProps): JSX.Element {
+    const [chosenDeck, setChosenDeck] = useState<Deck>();
+    const [decks, setDecks] = useImmer<Deck[]>([]);
+    const [chosenDecksCards, setChosenDecksCards] = useImmer<Card[]>([]);
+
+    const handleChooseDeck = (deckid: number) => {
+        const newDeck = decks.find((deck) => deck.deckid === deckid);
+        setChosenDeck(newDeck);
+    };
+
+    const fetchDecks = async () => {
+        const userDecks = await axios
+            .get<Deck[]>(`${baseUrl}/decks/${user.userid}`)
+            .then((response) => response.data);
+        setDecks(userDecks);
+    };
+
+    useEffect(() => {
+        fetchDecks();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    useEffect(() => {
+        fetchDecks();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [user]);
+
+    const fetchCards = async () => {
+        if (chosenDeck) {
+            const userCards = await axios
+                .get(`${baseUrl}/cards/${chosenDeck.deckid}`)
+                .then((response) => response.data);
+            setChosenDecksCards(userCards);
+        }
+    };
+
+    useEffect(() => {
+        fetchCards();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [chosenDeck]);
+
+    const addDeck = (deck: Deck) => {
+        setDecks((draft) => {
+            draft.push(deck);
+        });
+        setChosenDeck(deck);
+    };
+
+    const removeDeck = (deck: Deck) => {
+        setDecks((draft) => {
+            return draft.filter((d) => d.deckid !== deck.deckid);
+        });
+        setChosenDeck(undefined);
+    };
+
+    const addCardtoDeck = (card: Card) => {
+        setChosenDecksCards((draft) => {
+            draft.push(card);
+        });
+    };
```

Comment:
Nice idea to do this instead of rerendering with the new set of cards. Do you know the impact it has on performance?